src/adapter/src/error.rs

```diff
@@ -302,6 +303,9 @@ impl AdapterError {
                     earliest_possible,
                 ))
             }
+            AdapterError::UnallowedOnCluster { cluster, .. } => (cluster == MZ_INTROSPECTION_CLUSTER.name).then(||
+                format!("The transaction is executing on the {cluster} cluster, maybe having been routed there by the first statement in the transaction.")
```

Comment:
Is this too wordy? I'm not sure how to shorten it usefully.

Reply:
I think it's fine.